go/vt/vtgate/executor.go

```diff
@@ -229,18 +244,19 @@ func (e *Executor) handleExec(ctx context.Context, safeSession *SafeSession, sql
 			normalized := sqlparser.String(stmt)
 			sql = normalized + comments
 		}
-
+		logStats.PlanTime = execStart.Sub(logStats.StartTime)
 		logStats.SQL = sql
 		logStats.BindVariables = bindVars
 
-		execStart := time.Now()
-		logStats.PlanTime = execStart.Sub(logStats.StartTime)
+		var destination key.Destination
```

Comment:
These few lines can be merged with the 'if keyRange != nil' statement on line 228, so it's all done in one place.

Reply:
Done. Funny, I did it like that first, but then changed it. Thought it was better to separate validation from this logic.